tests/go-validation/tokenregistration/tokenregistration.go

```diff
@@ -0,0 +1,38 @@
+package tokenregistration
+
+import (
+	"github.com/rancher/norman/types"
+	v3Client "github.com/rancher/rancher/pkg/client/generated/management/v3"
+)
+
+type ClusterRegistrationToken struct {
+	v3Client.ClusterRegistrationTokenOperations
+}
+
+func NewClusterRegistrationToken(client *v3Client.Client) *ClusterRegistrationToken {
+	return &ClusterRegistrationToken{
+		client.ClusterRegistrationToken,
+	}
+}
+
+// CreateTokenRegistration is function that creates a ClusterRegistrationToken using a Client object with a specified *v3.ClusterRegistrationToken
+func (c *ClusterRegistrationToken) CreateTokenRegistration(clusterRegistrationToken *v3Client.ClusterRegistrationToken) (*v3Client.ClusterRegistrationToken, error) {
+	return c.Create(clusterRegistrationToken)
+}
+
+// GetTokenRegistration is function thatm gets a specific ClusterRegistrationToken using a Client object with a specified clusterStatusName and token name
+func (c *ClusterRegistrationToken) GetTokenRegistration(clusterId string) (*v3Client.ClusterRegistrationToken, error) {
```

Comment:
Nitpick: The function name seems reversed, I would think it should be called "GetRegistrationToken" instead.

Reply:
Fixed